packages/local-cli/link/__tests__/link-test.js

```diff
@@ -226,9 +226,76 @@ describe('link', () => {
 
     const link = require('../link').func;
 
-    link(['react-native-blur'], context).then(() => {
+    link(['react-native-blur'], context, {}).then(() => {
       expect(registerNativeModule.mock.calls).toHaveLength(1);
       done();
     });
   });
+
+  it('should link only for specific platforms if --platforms is used', async () => {
+    jest.setMock('../getProjectDependencies', () => ['react-native-maps']);
+    jest.setMock('../../core/getPackageConfiguration', () => ({
+      assets: [],
+    }));
+
+    const registerAndroidNativeModule = jest.fn();
+    const registerIOSNativeModule = jest.fn();
+
+    const genericAndroidLinkConfig = () => ({
+      isInstalled: () => false,
+      register: registerAndroidNativeModule,
+    });
+
+    const genericIOSLinkConfig = () => ({
+      isInstalled: () => false,
+      register: registerIOSNativeModule,
+    });
+
+    jest.setMock('../../core/getPlatforms', () => ({
+      android: { linkConfig: genericAndroidLinkConfig },
+      ios: { linkConfig: genericIOSLinkConfig },
+    }));
+
+    jest.setMock(
+      '../android/registerNativeModule.js',
+      registerAndroidNativeModule
+    );
+    jest.setMock('../ios/registerNativeModule.js', registerIOSNativeModule);
+
+    const link = require('../link').func;
+    const assertPlaftormsCalledTimes = (android, ios) => {
+      expect(registerAndroidNativeModule).toHaveBeenCalledTimes(android);
+      expect(registerIOSNativeModule).toHaveBeenCalledTimes(ios);
+      registerAndroidNativeModule.mockClear();
+      registerIOSNativeModule.mockClear();
+    };
+
+    await link(
+      ['react-native-gradient'],
+      { root: '/' },
+      { platforms: ['android', 'ios'] }
+    );
+    assertPlaftormsCalledTimes(1, 1);
+
+    await link(
+      ['react-native-gradient'],
+      { root: '/' },
+      { platforms: ['android'] }
+    );
+    assertPlaftormsCalledTimes(1, 0);
+
+    await link(
+      ['react-native-gradient'],
+      { root: '/' },
+      { platforms: ['ios'] }
+    );
+    assertPlaftormsCalledTimes(0, 1);
+
+    await link(
```

Comment:
this link call is a duplicate of https://github.com/react-native-community/react-native-cli/pull/77/files?w=1#diff-f71ce3341c36eb9ac6eefd5606744ff6R273, can you remove it?

Reply:
Oh you are right! One of them should be empty array.